modules/packages/ZMQ.chpl

```diff
@@ -728,6 +734,77 @@ module ZMQ {
       }
     }
 
+    /*
+      Get the last endpoint for the specified socket; see
+      `zmq_getsockopt <http://api.zeromq.org/4-0:zmq-getsockopt>`_ under
+      ZMQ_LAST_ENDPOINT.
+
+      :returns: The last endpoint set, see the link above.
+      :rtype: string
+    */
+    proc getLastEndpoint(): string {
+      var ret: string;
+      on classRef.home {
+        var str: c_string;
+        var err = zmq_getsockopt_string_helper(classRef.socket,
+                                               ZMQ_LAST_ENDPOINT, str);
+        if err == -1 {
+          var errmsg = zmq_strerror(errno):string;
+          // It would be good to throw an Error instead, see issue #12397 on
+          // Github
+          halt("Error in Socket.getLastEndpoint(): ", errmsg);
```

Comment:
I don't think we should merge this with a `halt` in it. At the very least if you create a `ZMQError` (or `ZmqError`?) it can store the string result of `zmq_strerror`. I think it would be interesting to also consider how that might merge with SystemError, but I think having these functions throw is important from the get-go.